### ISSUE TITLE
facenet input shape problem

### ISSUE THREAD

**Issue by breakdown-hh** [NVIDIA/tensorrt-inference-server#561]
hi, google facenet pre-trained model has two input tensor, one is the input image of dim[160, 160, 3], another is a flag called phase_train, indicates whether it is training phase or inference phase, and the type is bool, it's a scalar.

The TensorRT Inference Server seems to force the input parameter to be a tensor array，and it also has a reshape option that can change the dim of input tensor when the tensor is submitted to the backend. 

However, when I configure the config.pbtxt like below:

name: "facenet_savedmodel"
platform: "tensorflow_savedmodel"
max_batch_size: 128
input [
{
name: "input"
data_type: TYPE_FP32
format: FORMAT_NHWC
dims: [ 160, 160, 3 ]
},
{
name: "phase_train"
data_type: TYPE_BOOL
dims: [ 1 ]
reshape: { shape: [ ] }
}
]

output [
{
name: "embeddings"
data_type: TYPE_FP32
dims: [ 512 ]
}
]

I got the error below:
tensorrtserver.api.InferenceServerException: [inference:0 271] The second input must be a scalar, but it has shape [1]

How to solve this problem? thanks.

**Comment by GuanLuo** [comment]
Currently TRTIS expects the inputs to be tensors. Reshape to [] doesn't actually produce a scalar, instead, it produce a tensor with one less dimension. For instance, your model config is actually reshaping from [ batch_dim, 1 ] to [ batch_dim ]. To solve this problem, you will need to modify your model to always feed a constant value to "phase_train", and thus it is no longer an input needed from TRTIS.

And do you mind to share the model or where to get it?

**Comment by breakdown-hh** [comment]
@GuanLuo thanks for reply, the pre-trained model is from google facenet.
https://github.com/davidsandberg/facenet
pre-trained models: 20180408-102900 and 20180402-114759

**Comment by bobeu** [comment]
Will add some update shortly

**Comment by deadeyegoodwin** [comment]
Closing due to no response. Re-open if you have more information.
